servlet/src/main/java/io/undertow/servlet/spec/HttpServletRequestImpl.java

```diff
@@ -255,10 +255,12 @@ public HttpServletMapping getHttpServletMapping() {
                 }
                 break;
             case EXTENSION:
-                matchValue = match.getMatched().substring(0, match.getMatched().length() - match.getMatchString().length() + 1);
-                if(matchValue.startsWith("/")) {
-                    matchValue = matchValue.substring(1);
-                }
+                String remaining = match.getRemaining();
+                String matched = remaining != null ? remaining : match.getMatched();
+                String matchString = match.getMatchString();
+                int startIndex = matched.startsWith("/") ? 1 : 0;
+                int endIndex = matched.length() - matchString.length() + 1;
+                matchValue = matched.substring(startIndex, endIndex);
```

Comment:
I combined the two subsequent substring operations to avoid re-allocating the matchValue string.